### ISSUE TITLE
Pack static assets into the deployed binary

### ISSUE THREAD

**Issue by cweagans** [adtac/commento#131]
It would be awesome to be able to deploy commento as a single binary with all the assets packed inside.

https://github.com/jteeuwen/go-bindata and https://github.com/elazarl/go-bindata-assetfs would make that possible, but we'd need to figure out how to make it not get in the way of development. Maybe it can just be something that happens in a release process?

**Comment by lrstanley** [comment]
I'd suggest using https://github.com/GeertJohan/go.rice rather than go-bindata for the sole purpose of how go.rice is easy when developing.

1. It allows the application to compile, without having to generate files first. Meaning code completion still works properly, and it doesn't complain that variables don't exist (unlike go-bindata, where you have to generate the bindata file first.)
2. If the the bindata file hasn't been generated yet and you compile the Go app, it will also look for the assets on disk, meaning you don't _have_ to bindata then recompile each time you make a change, and only have to do that when you distribute.
3. Provides three different options of how to add the data to the binary (this may not be super important, because the most standard way is what many bindata-style utilities support.)

tl;dr I loved go-bindata originally, because it allowed me to bundle the assets with the application, however it made development a HUGE pain, having to regenerate the bindata files each time I make a change and to get my IDE/editor from going nuts because variables didn't exist. go.rice solved that for me.

If you take a look at my personal sites [Makefile](https://github.com/lrstanley/liam.sh/blob/master/Makefile#L18-L33), you will see that I have simple tasks to:

1. Make sure that the go.rice tool is installed.
2. Clean any bindata files if they exist (i.e. to make sure they don't get used when debugging)
3. If debugging, clean and run. If building, clean, generate bindata, then compile.

Also note that there is no "assetfs" based package for go.rice, since I believe everything that the package in mention does, go.rice has support for out of the box. Hope this helps!

**Comment by lrstanley** [comment]
I should also mention that the other benefit is that you also don't really need to check in the bindata-generated file to git, because it can work from both disk, and from the file. This is also beneficial because it cuts down on clutter in git history and similar, for when any assets are updated. If one was to use go-bindata, you would likely need to check it in.

**Comment by cweagans** [comment]
Updated the title to reflect my actual intent. I don't have strong feelings about which tool gets used, but I do think packing the assets into the final binary would be awesome.

**Comment by adtac** [comment]
@cweagans @lrstanley @mro - Thanks a lot for this issue! This issue is still very much possible.

We have, however, moved all development [to GitLab](https://gitlab.com/commento/commento-ce). Therefore, this comment has been merged with https://gitlab.com/commento/commento-ce/issues/3 (with a link back to this issue for history). Go.rice seems to be the popular opinion :)
